gcc/rust/util/rust-inline-visitor.h

```diff
@@ -0,0 +1,63 @@
+/* An implementation of the inline visitor.
+   https://members.accu.org/index.php/articles/2021 */
+
+#include <utility>
+
+#ifndef RUST_INLINE_VISITOR
+#define RUST_INLINE_VISITOR
+
+namespace Rust {
+
+/* VisitorWrapper is a wrapper over a (possibly incomplete) Visitor.
```

Comment:
Style nit here: I prefer comments to be // for multiline. Clang format helps here.